pyvista/core/filters/poly_data.py

```diff
@@ -2601,3 +2601,114 @@ def strip(poly_data, join=False, max_length=1000, pass_cell_data=False,
         alg.SetPassThroughPointIds(pass_point_ids)
         _update_alg(alg, progress_bar, 'Stripping Mesh')
         return _get_output(alg)
+
+    def collision(poly_data, other_mesh, contact_mode=0, box_tolerance=0.001,
+                  cell_tolerance=0.0, n_cells_per_node=2, generate_scalars=False,
+                  progress_bar=False):
+        """Perform collision determination between two polyhedral surfaces.
+
+        If ``collision_mode`` is set to all contacts, the output will
+        be lines of contact. If ``collision_mode`` is first contact or half
+        contacts then the Contacts output will be vertices.
+
+        Parameters
+        ----------
+        other_mesh : pyvista.PolyData
```

Comment:
Does it make sense to check that `isinstance(other_mesh, pyvista.PolyData)`?

Reply:
Yes, and out of convenience, I'll extract the surface if it's not polydata.

Reply:
Actually, according to [vtkCollisionDetectionFilter_8h_source.html](https://vtk.org/doc/nightly/html/vtkCollisionDetectionFilter_8h_source.html), I'll need to check if the meshes are triangular as well.

Reply:
Added a check and convert.